src/puppetlabs/puppetdb/query_eng.clj

```diff
@@ -24,33 +25,35 @@
    currently adhere to that contract, so this function will wrap the
    given function `f` and ignore those arguments"
   [f]
-  (fn [_ _]
+  (fn [_ _ _ _]
     f))
 
 (defn stream-query-result
   "Given a query, and database connection, return a Ring response with the query
   results.
+
   If the query can't be parsed, a 400 is returned."
-  [entity version query paging-options db output-fn]
+  [entity version query paging-options db base-url output-fn]
```

Comment:
So any reason for using base-url here instead of just passing url-prefix through? Just passing url-prefix and version through would save a lot of duplication.